crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -107,7 +107,7 @@ module sui::validator_set {
         ctx: &TxContext,
     ) {
         let validator_address = tx_context::sender(ctx);
-        let validator = get_validator_mut(&mut self.active_validators, validator_address);
+        let validator = vec_map::get_mut(&mut self.active_validators, &validator_address);
```

Comment:
This is interesting. I was wondering since `vec_map::get_mut` has an `&K` if you would run into the same "lifetime" lossyness that we saw in the Table API.

But I suppose that if you require `K` to have `copy` it should be easy enough to generate a temp ref as was done here (and since it will be released after the call, no lifetime confusion issue). Just a possible point to consider when thinking about a native `map` API down the road.

Reply:
I wouldn't be opposed to asking for ownership of `K`--the references are kind of annoying, and it's not clear that they actually create an efficiency gain.